### PR TITLE
Bowtie2: add sort option

### PR DESCRIPTION
**Pull request by matthdsm**
add samtools sort option cfr bwa/bwa-mem2

## PR checklist

Closes #XXX 

- [ ] This comment contains a description of changes (with reason).
- [ ] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [ ] Follow the parameters requirements.
- [ ] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [ ] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - [ ] `PROFILE=docker pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=singularity pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`
  - [ ] `PROFILE=conda pytest --tag <MODULE> --symlink --keep-workflow-wd --git-aware`

### CODE REVIEW

File: tests/modules/bowtie2/align/test.yml [hunk 1]
```diff
@@ -8,6 +8,16 @@
...
+- name: bowtie2 align test_bowtie2_align_single_end_sorted
+  command: nextflow run ./tests/modules/bowtie2/align -entry test_bowtie2_align_single_end_sorted -c ./tests/config/nextflow.config -c ./tests/modules/bowtie2/align/nextflow.config
+  tags:
+    - bowtie2
+    - bowtie2/align
+  files:
+    - path: ./output/bowtie2/test.bam
+    - path: ./output/bowtie2/test.bowtie2.log
+    - path: ./output/bowtie2/versions.yml
+
```
Comment (JoseEspinosa) [inline tests/modules/bowtie2/align/test.yml:20]: Maybe also adding a test for paired-end data as in the [bwa module](https://github.com/nf-core/modules/blob/919275103a00ddbcfc31b2c424837efaaf2ef0a8/tests/modules/bwa/mem/main.nf#L60-L72)

**Review by JoseEspinosa — CHANGES_REQUESTED** [review]
Thanks @matthdsm just a small suggestion otherwise looks awesome!

File: tests/modules/bowtie2/align/test.yml (hunk 1 above)
Comment (matthdsm) [reply tests/modules/bowtie2/align/test.yml:20]: done!

**Review by JoseEspinosa — APPROVED** [review]
Awesome @matthdsm :rocket: 
Thanks!

**Pull request merged**